update translation guide (#19099)
gatsbyjs/gatsby: docs/contributing/gatsby-docs-translation-guide.md

The first step for starting a new translation is to check what exists. So far, there are repositories for the following languages:

- [German](https://github.com/gatsbyjs/gatsby-de)

> - [Spanish](https://github.com/gatsbyjs/gatsby-es)

Author: Rearranged based on language code to be more neutral (same thing we do for React)

- [Indonesian](https://github.com/gatsbyjs/gatsby-id)
- [Italian](https://github.com/gatsbyjs/gatsby-it)
- [Korean](https://github.com/gatsbyjs/gatsby-ko)
- [Dutch](https://github.com/gatsbyjs/gatsby-nl)
- [Polish](https://github.com/gatsbyjs/gatsby-pl)
- [Brazilian Portuguese](https://github.com/gatsbyjs/gatsby-pt-BR)
- [Russian](https://github.com/gatsbyjs/gatsby-ru)
- [Turkish](https://github.com/gatsbyjs/gatsby-tr)

> Note: Once a new translation repository is created, feel free to add it here in a PR!

If your language is not listed above, check the list of [open translation requests](https://github.com/gatsbyjs/gatsby/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+%22New+Translation+Request%22).

> If you don't see the language among the issues listed, feel free to creat a new [translation request issue](https://github.com/gatsbyjs/gatsby/issues/new?template=new_translation.md) for it and follow the instructions.

Reviewer: ```suggestion
If you don't see the language among the issues listed, feel free to create a new [translation request issue](https://github.com/gatsbyjs/gatsby/issues/new?template=new_translation.md) for it and follow the instructions.
```

- Act as point of contact for your language and answer questions from both contributors to your language and the core Gatsby team.
- Set up a process in order to get your translation published. (details to come)

As a maintainer, you are welcome to add a contributing doc written in your language to assist with the process. You can find an example in the [gatsby-es repo](https://github.com/gatsbyjs/gatsby-es/blob/master/CONTRIBUTING.MD). Translating [this page](https://github.com/gatsbyjs/gatsby/blob/master/docs/contributing/gatsby-docs-translation-guide.md) and copying it into a `contributing.md` file would be an option as well.

### Additional tips

#### Set up a style guide and glossary

Your language repo comes with a template [style guide](https://github.com/gatsbyjs/gatsby-i18n-source/blob/master/style-guide.md) that you can use to put in style rules specific to your language. Refer to the [translation style guide](#translation-style-guide) section for more information.

#### Collaborate on difficult pages

Some pages are very long and difficult to translate for a single contributor. Feel free to split these pages up into sections and work on them together!

> In addition, sometimes a contributor cannot continue a PR for whatever reason. Feel free to ask them if they need help and to bring someone else in to finish the work for them!

Reviewer: Changing up the wording a bit:
```suggestion
In addition, sometimes a contributor cannot continue a PR for whatever reason. It may help to ask them if they need assistance and to bring someone else on to complete the work in a timely fashion.
```

#### Set up a review process

As codeowners, you have the freedom and responsibility to decide what your review process will be like. You can decide how many reviewers you'd like. If your team is small, one reviewer may be enough. But if you have lots of contributors and enough codeowners, you may want to require two reviewers for additional quality.

> Feel free to install any plugin or automation that will make your life easier as codeowners. Is there a text linter that works well in your language? Is there any automation that you can add? If you feel that these improvements would be beneficial to other languages as well, feel free to create an issue or PR for it in the main Gatsby repository.

Reviewer: ```suggestion
You have the ability to install any plugin or automation tool that will make your life easier as codeowners. Is there a text linter that works well in your language? Is there any automation that you can add? If you feel that these improvements would be beneficial to other languages as well, create an issue or PR for it in the main Gatsby repository.
```

> You have the ability to install any plugin or automation tool that will make your life easier as codeowners. Is there a text linter that works well in your language? Is there any automation that you can add? If you feel that these improvements would be beneficial to other languages as well, create an issue or PR for it in the main Gatsby repository.

Reviewer: One question: are there any limits on what they can install?

Author: Does gatsby have any restrictions on what apps can be installed or what packages can be added as dependencies? Also, any new github plugins have to go through the gatsby admins before they get installed.

I guess I should change this to:

> You are empowered to install plugins and automation tools that will make your life easier as codeowners. [...] Create an issue/PR and tag @gatsby-core so we can review the use case. If you feel that these improvements would be beneficial...

Reviewer: Those are mostly questions for @gatsbyjs/core in terms of package/app restrictions, but that seems like a good idea to have contributors open an issue and tag @gatsby-core to vet stuff.

Reviewer: We can merge this in as-is and make changes to it later if there ends up being things requested that we don't want to approve. Thanks for adding this in there!

#### Ask for help

> Don't be afraid to ask for help! If you're not sure about something, feel free to post in the `#localization` channel on the [Gatsby Discord](https://gatsby.dev/discord) or create an issue in the Gatsby repo.

Reviewer: ```suggestion
Don't be afraid to ask for help! If you're not sure about something, you can post in the `#localization` channel on the [Gatsby Discord](https://gatsby.dev/discord) or create an issue in the Gatsby repo.
```

> If it feels like there is too much work and you need help, feel free to add more codeowners by editing the `CODEOWNERS` file in the repo. Are there any contributors who are making exceptional contributions? If so, consider making them a codeowner.

Reviewer: ```suggestion
If it feels like there is too much work and you need help, you have the ability to to add more codeowners by editing the `CODEOWNERS` file in the repo. Are there any contributors who are making exceptional contributions? If so, consider making them a codeowner.
```

### Glossary

The style guide has a [glossary section](https://github.com/gatsbyjs/gatsby-i18n-source/blob/master/style-guide.md#glossary) that you can use to fill in common translations. Look at the English [Glossary](/docs/glossary/) for a list of terms that are useful to have translations for.

### Universal style guide

The following rules should apply in all translations and can serve as a basis for your language-specific style guide.

#### Keep the meaning of the source

Keep the meaning of the original English source even if it is confusing or has a typo. If you find an error that can be fixed, create an issue or pull request to the original [gatsby repo](https://github.com/gatsbyjs/gatsby) so that all translations can benefit from the change.

#### Text in code blocks

Leave text in code blocks untranslated except for comments. You may optionally translate text in strings, but be careful not to translate strings that refer to code!

Example:

> ```js

Reviewer: It seems like we could skip these first examples and go straight to the the Do's and Don'ts. What do you think?

)
```

❌ DEFINITELY DON'T:

```js
importar Reaccionar desde "reaccionar"
exportar defecto () => (
   <div estilo = {{color: `morado`, fontSize:` 72px`}}> ¡Hola Gatsby! </div>
)
```

#### Internal links

Translate link text but keep all slugs and hashes in links the same as they are in English.

Example:

```md
- [Configure your development environment](/tutorial/set-up-your-development-environment)
```

✅ OK:

```md
- [Configure su entorno de desarrollo](/tutorial/set-up-your-development-environment)
```

❌ DON'T:

```md
- [Configura tu entorno de desarrollo](/tutorial/configura-tu-entorno-de-desarrollo)
```

#### External links

If an external link is to an article in a reference like [MDN] or [Wikipedia], and a version of that article exists in your language that is of decent quality, consider linking to that version instead.

[mdn]: https://developer.mozilla.org/en-US/
[wikipedia]: https://en.wikipedia.org/wiki/Main_Page

Example:

```md
React elements are [immutable](https://en.wikipedia.org/wiki/Immutable_object).
```

✅ OK:

```md
Los elementos de React son [inmutables](https://es.wikipedia.org/wiki/Objeto_inmutable).
```

> For links that have no equivalent (Stack Overflow, YouTube videos, etc.), just use the English link.

Reviewer: ```suggestion
For links that have no equivalent (Stack Overflow, YouTube videos, etc.), use the English link.
```
